config/src/test/scala/com/typesafe/config/impl/PublicApiTest.scala

```diff
@@ -17,6 +16,15 @@ import java.net.URL
 import java.time.Duration
 
 class PublicApiTest extends TestUtils {
+
+    @Before
+    def before(): Unit = {
+        // TimeZone.getDefault internally invokes System.setProperty("user.timezone", <default time zone>) and it may
+        // cause flaky tests depending on tests order and jvm options. This method is invoked
+        // eg. by URLConnection.getContentType (it reads headers and gets default time zone).
```

Comment:
Yes, it took few hours. I had to use windows OS because on linux I wasn't able to reproduce the issue - it was horrible :)